### PR TITLE
fixed $archive and $agent_installer references, refactored archive name

### PR DESCRIPTION
**Pull request by timidri**
Fixed some variable mixups and dependencies

### CODE REVIEW

File: manifests/agent/nix.pp [hunk 1]
```diff
@@ -16,37 +16,37 @@
...
 
-  archive { "${homedir}/${archive}" :
+  archive { "${homedir}/${archive}.gz" :
     source       => $url,
     user         => 'distelli',
     group        => 'distelli',
     extract      => true,
     extract_path => $homedir,
-    creates      => "${homedir}/distelli",
+    creates      => "${homedir}/${archive}",
```
Comment (abrader) [inline manifests/agent/nix.pp:39]: @timidri the issue is the creates doesn't create the archive when the extract => true attribute is set, it creates what was in the archive.  in this case it is a folder called distelli.

Comment (timidri) [reply manifests/agent/nix.pp:39]: @abrader Hmm, let me retest for idempotency

Comment (timidri) [reply manifests/agent/nix.pp:39]: @abrader Actually, it works fine. The `creates` doesn't actually create anything, it just prevents `archive` from executing again when the file mentioned in creates already exists. The archive doesn't contain a folder. Log of the puppet run on a CentOS 7 machine:
```
[root@centos7a distelli]# puppet agent -t
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for centos7a.pdx.puppet.vm
Info: Applying configuration version '8ca21ad459581895787263a1b110b46806c0d1cb'
Notice: /Stage[main]/Distelli::Agent::Nix/Archive[/home/distelli/distelli.Linux-x86_64-3.66.33.gz]/ensure: download archive from https://s3.amazonaws.com/download.distelli.com/distelli.Linux-x86_64/distelli.Linux-x86_64-3.66.33.gz to /home/distelli/distelli.Linux-x86_64-3.66.33.gz and extracted in /home/distelli with cleanup
Notice: /Stage[main]/Distelli::Agent::Nix/File[/home/distelli/distelli.Linux-x86_64-3.66.33]/mode: mode changed '0644' to '0755'
Notice: /Stage[main]/Distelli::Agent::Nix/File[/home/distelli/distelli]/ensure: created
Info: /Stage[main]/Distelli::Agent::Nix/File[/home/distelli/distelli]: Scheduling refresh of Exec[Test agent executable]
Notice: /Stage[main]/Distelli::Agent::Nix/Exec[Test agent executable]: Triggered 'refresh' from 1 event
Info: /Stage[main]/Distelli::Agent::Nix/Exec[Test agent executable]: Scheduling refresh of Exec[Install agent]
Notice: /Stage[main]/Distelli::Agent::Nix/Exec[Install agent]: Triggered 'refresh' from 1 event
Notice: Applied catalog in 6.28 seconds
```

Check for idempotency:

```
[root@centos7a distelli]# puppet agent -t
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Info: Caching catalog for centos7a.pdx.puppet.vm
Info: Applying configuration version '8ca21ad459581895787263a1b110b46806c0d1cb'
Notice: Applied catalog in 1.84 seconds
```

**Review by abrader — APPROVED** [review]
Tested.  Works well.

**Pull request merged**